src/trie_tree.rs

```diff
@@ -0,0 +1,456 @@
+                                    proof_result.push(
+                                        MergeValue::shortcut(key, value, fork_height)
+                                            .into_merge_with_zero::<H>(),
+                                    );
+                                }
+                                if fork_height == 1 && leaves_bitmap[leaf_index].get_bit(0) {
+                                    proof_result.push(MergeValue::from_h256(value));
+                                }
+                            }
+                            break;
+                        }
+                    } else {
+                        // Maybe we've skipped shortcut node, find from up to down
+                        for i in (height..=core::u8::MAX).rev() {
+                            let parent_key = leaf_key.parent_path(i);
+                            let is_right = leaf_key.is_right(i);
+                            let parent_branch_key = BranchKey::new(i, parent_key);
+                            if let Some(parent_branch) =
+                                self.store.get_branch(&parent_branch_key)?
+                            {
+                                let current = if is_right {
+                                    parent_branch.right
+                                } else {
+                                    parent_branch.left
+                                };
+
+                                match current {
+                                    MergeValue::ShortCut { key, value, .. } => {
+                                        if !key.eq(&leaf_key) {
+                                            let fork_at = key.fork_height(&leaf_key);
+                                            if fork_at == height {
+                                                proof_result.push(
+                                                    MergeValue::shortcut(key, value, height)
+                                                        .into_merge_with_zero::<H>(),
+                                                );
+                                            }
+                                        }
+                                        break;
+                                    }
+                                    _ => {
+                                        continue;
+                                    }
+                                }
+                            }
+                        }
+                        break; // we should stop further looping
+                    }
+                }
+            }
+
+            proof_result.reverse();
+            proof.append(&mut proof_result);
+            debug_assert!(stack_top < MAX_STACK_SIZE);
+            stack_fork_height[stack_top] = fork_height;
+            stack_top += 1;
+            leaf_index += 1;
+        }
+        Ok(MerkleProof::new(leaves_bitmap, proof))
+    }
+}
+
+/// Helper function for a merkle_path insertion
```

Comment:
Rust doc comment is for public function, use `//` instead.

Reply:
FWIW, there's `document-private-items`: https://doc.rust-lang.org/cargo/commands/cargo-doc.html#documentation-options